src/Event/ElasticsearchOperationErrorEvent.php

```diff
@@ -58,12 +65,14 @@ class ElasticsearchOperationErrorEvent extends Event {
    * @param \Throwable $error
    * @param $operation
    * @param \Drupal\elasticsearch_helper\Plugin\ElasticsearchIndexInterface $plugin_instance
+   * @param mixed|null $object
    * @param \Drupal\elasticsearch_helper\ElasticsearchRequestWrapperInterface $request_wrapper
    */
-  public function __construct(\Throwable $error, $operation, ElasticsearchIndexInterface $plugin_instance, ElasticsearchRequestWrapperInterface $request_wrapper = NULL) {
+  public function __construct(\Throwable $error, $operation, ElasticsearchIndexInterface $plugin_instance, $object = NULL, ElasticsearchRequestWrapperInterface $request_wrapper = NULL) {
```

Comment:
It looks like at least that the Request wrapper is consistently passed to the event object. Perhaps it's best if $object would be the last argument in the constructor, so that we could optionally pass it only when needed instead of setting it as Null in several places.

Reply:
There are more cases where `$request_wrapper` is present whereas `$source` is not, rather than the other way around. For `drop` operation there's never going to be `$source` available, but high probability that `$request_wrapper` is going to be provided.